### PR TITLE
init: setup package structure

### PR DESCRIPTION
**Pull request by rekt-hard**
Initial implementation of curations.

### CODE REVIEW

File: invenio_curations/services/service.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            ),
+        )
+        return next(results.hits) if results.total > 0 else None
+
+    @unit_of_work()
+    def create(self, identity, data=None, uow=None, **kwargs):
+        """Creates a RDMCuration request and submits it."""
+
+        role = self.role()
+        assert role, _(
```
Comment (utnapischtim) [inline invenio_curations/services/service.py:81]: question: could an `assert` not be removed with an optimization flag? would it not be better to use
```
if not role: 
    raise RuntimeError()
```

or something like that?

File: invenio_curations/views/ui.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        "invenio_curations",
+        __name__,
+        template_folder="../templates",
+        static_folder="../static",
+    )
+
+    # Add URL rules
+    blueprint.add_url_rule("/curations/overview", view_func=curation_requests_overview)
+
+    @blueprint.before_app_first_request
```
Comment (utnapischtim) [inline invenio_curations/views/ui.py:57]: Major: please use the new `finalize_app` entrypoint and not the deprecated `before_app_first_request`

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+[options]
+include_package_data = True
+packages = find:
+python_requires = >=3.9
+zip_safe = False
+install_requires =
+    invenio-requests>=4.0.0,<5.0.0
+[options.extras_require]
+tests =
+    pytest-black>=0.3.0
```
Comment (utnapischtim) [inline setup.cfg:31]: change it please to `pytest-black-ng>=0.4.0` pytest-black is unsupported

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    invenio-app>=1.3.2,<2.0.0
+    invenio-db[postgresql,mysql]>=1.0.13,<2.0.0
+    pytest-invenio>=2.1.0,<3.0.0
+    sphinx>=4.5.0
+elasticsearch7 =
```
Comment (utnapischtim) [inline setup.cfg:36]: in my opinion elasticsearch7 is not more necessary. it has been removed from invenio packages too as i know

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+[pydocstyle]
+add_ignore = D401
+
+[compile_catalog]
+directory = invenio_curations/translations/
+use-fuzzy = True
+
+[extract_messages]
+copyright_holder = CERN
```
Comment (utnapischtim) [inline setup.cfg:77]: is CERN the copyright holder or TU Graz?

File: invenio_curations/services/service.py (hunk 1 above)
Comment (rekt-hard) [reply invenio_curations/services/service.py:81]: Good catch. Will update

File: invenio_curations/views/ui.py (hunk 2 above)
Comment (rekt-hard) [reply invenio_curations/views/ui.py:57]: Good catch. Will update

**Pull request merged**